ui/page/start_page.go

```diff
@@ -600,16 +613,40 @@ func (sp *startPage) currentPageIndicatorLayout(gtx C) D {
 	})
 }
 
-func (sp *startPage) setLanguageSetting() {
-	langPre := sp.AssetsManager.GetLanguagePreference()
-	if langPre == "" {
-		sp.AssetsManager.SetLanguagePreference(values.DefaultLanguage)
+func (sp *startPage) setLanguagePref(useExistingUserPreference bool) {
+	var lang string
+	if useExistingUserPreference {
+		lang = sp.AssetsManager.GetLanguagePreference()
+	} else {
+		lang = sp.selectedLanguageKey()
+	}
+	if lang == "" {
+		lang = values.DefaultLanguage
+	}
+	sp.AssetsManager.SetLanguagePreference(lang)
+	values.SetUserLanguage(lang)
+}
+
+func (sp *startPage) selectedLanguageKey() string {
+	selectedLang := sp.languageDropdown.Selected()
+	for _, opt := range preference.LangOptions {
+		if strings.ToLower(selectedLang) == opt.Value {
+			return opt.Key
+		}
 	}
-	values.SetUserLanguage(langPre)
+	return values.DefaultLanguage
 }
 
-//func (sp *startPage) recommendedSettings() {
-// 	To be implemented after settings page refactor
-// Should set settings for USD exchange, Fee rate api,
-// exchange api, and transaction notifications to enabled.
-//}
+func (sp *startPage) updateSettings() {
+	wantAdvanced := sp.selectedSettingsOptionIndex == advancedSettingsOptionIndex
+	if wantAdvanced {
+		return // nothing to do?
```

Comment:
we should probably open a settings page and allow then pick or click done

let's discuss this with the designers

Reply:
Okay.